lib/client.js

```diff
@@ -55,7 +55,12 @@ function Client(config = {}) {
     this.version = config.version;
   }
 
-  this.connectionSettings = cloneDeep(config.connection || {});
+  if (config.connection && config.connection instanceof Function) {
+    this.connectionConfigProvider = config.connection;
+    this.connectionSettings = { expired: () => true }; // causes the provider to be called on first use
```

Comment:
`expirationChecker` might be a more self-explanatory name, to emphasize that it's not a flag.

Reply:
will change